pkgs/applications/misc/ironbar/default.nix

```diff
@@ -0,0 +1,83 @@
+{ gtk3
+, gdk-pixbuf
+, librsvg
+, webp-pixbuf-loader
+, gobject-introspection
+, glib-networking
+, glib
+, shared-mime-info
+, gsettings-desktop-schemas
+, wrapGAppsHook
+, gtk-layer-shell
+, gnome
+, libxkbcommon
+, openssl
+, pkg-config
+, hicolor-icon-theme
+, rustPlatform
+, lib
+, fetchFromGitHub
+}:
+
+rustPlatform.buildRustPackage {
+  pname = "ironbar";
+  version = "0.12.1-unstable-2023-07-04";
+
+  src = fetchFromGitHub {
+    owner = "JakeStanger";
+    repo = "ironbar";
+    rev = "f8d8c063008340922b20aa665d8f893ab63e82d7";
+    sha256 = "sha256-lObK6FchYCvHgF/wIxzNbdC/W8xqR+7kh59jMEyKgv0=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes."stray-0.1.3" = "sha256-7mvsWZFmPWti9AiX67h6ZlWiVVRZRWIxq3pVaviOUtc=";
+  };
+
+  buildInputs = [
+    gtk3
+    gdk-pixbuf
+    glib
+    gtk-layer-shell
+    glib-networking
+    shared-mime-info
+    gnome.adwaita-icon-theme
+    hicolor-icon-theme
+    gsettings-desktop-schemas
+    libxkbcommon
+    openssl
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
+    wrapGAppsHook
+    gobject-introspection
+  ];
+
+  propagatedBuildInputs = [
+    gtk3
+  ];
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      # Thumbnailers
+      --prefix XDG_DATA_DIRS : "${gdk-pixbuf}/share"
+      --prefix XDG_DATA_DIRS : "${librsvg}/share"
+      --prefix XDG_DATA_DIRS : "${webp-pixbuf-loader}/share"
+      --prefix XDG_DATA_DIRS : "${shared-mime-info}/share"
+    )
+  '';
```

Comment:
These were added as part of a fix for blurry images.  For flake/home manager at least, setting them did have an impact.  https://github.com/JakeStanger/ironbar/pull/122

Unsure as to what the expected behaviour here is.

Reply:
> These were added as part of a fix for blurry images.  For flake/home manager at least, setting them did have an impact.  https://github.com/JakeStanger/ironbar/pull/122
> 
> Unsure as to what the expected behaviour here is. 

The fix was to add the GTK wrapper, but the example for the wrapper on some docs (I think nixpkgs) showed it something like this.

Reply:
I'll test without this part later to see if I notice a difference.

Reply:
@yavko So I don't notice a difference but I don't use hidpi. Can you confirm that this is still needed? If so, I'll need to figure something out to avoid the double wrapping with `wrapProgram` in `postInstall`.

Reply:
Double wrapper should be fixed now (will assume that this is needed)